### DISCUSSION TITLE
Collecting ideas for right-click on route

### DISCUSSION THREAD

**Discussion by moving-bits** [cgeo/cgeo#13958] [Ideas]
Coming from the discussion in https://github.com/cgeo/cgeo/pull/13911#issuecomment-1416782282 let's collect and discuss ideas for what to do on "tap on route".

Some ideas: Open a selection popup with following options:
- hide/unhide route
- sort route
- reverse route
- export route
- open route/track quick menu
- mark last entry as visited and remove from route

Most of this is already available via the route/track quick settings button, so I'm not sure if it's really needed as "tap on route" action.

Only really new thing is the last entry. But this one is actually bound to a route's point, so it should rather go into the point's context menu than in the route's one, IMHO.

But there may be other, more useful ideas / use-cases. Feel free to add...

**Comment by fm-sys** [comment]
> Most of this is already available via the route/track quick settings button, so I'm not sure if it's really needed as "tap on route" action. 

These actions do all refer to one very special route type. c:geo's individual route. 

The main benefit will come when several different GPX tracks are loaded into the map. Making the track clickable allows hiding/removing tracks even if you don't know there specific name.

**Comment by fm-sys** [comment]
`UnifiedMapActivity` has a very basic abstraction layer:

- `UnifiedMapActivity` calls its current map view provider to retrieve a list of affected items. If it's more than one, you get a popup selection menu.
- [...] 

By using this method I was able to provide a standardized popup selection for both VTM and Google Maps (a feature not yet implemented in today's GMv2 map).

_Originally posted by @moving-bits in https://github.com/cgeo/cgeo/issues/13911#issuecomment-1416809506_
        
---
   
Maybe routes/tracks could even be included into this selection popup? 

Locus is doing this BTW...

|||
|-|-|
![Screenshot_20230205_000853_menion android locus pro computerBild](https://user-images.githubusercontent.com/64581222/216793371-fa114982-7080-4d6b-920a-89f835f0acae.jpg)|![Screenshot_20230205_000858_menion android locus pro computerBild](https://user-images.githubusercontent.com/64581222/216793373-523d31c8-8da8-4386-a567-f93ba788d6a2.jpg)
